### PR TITLE
Cloud DNS ManagedZone IAM resources

### PR DESCRIPTION
**Pull request by hao-nan-li**
Add Cloud DNS ManagedZone IAM resources

Fixed https://github.com/hashicorp/terraform-provider-google/issues/12906

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
Add Cloud DNS ManagedZone IAM Resource
```

### CODE REVIEW

**Comment by hao-nan-li** [comment]
Note to the reviewer: I got the following error: **"Please verify ownership of the 'googlecloudexample.com.' domain (or a parent) at http://www.google.com/webmasters/verification/ and try again"** locally. I want to see if this is per project or per account or something else, so attempted to run the tests here. Looks like it's re-produced. 

I posted the error on https://b.corp.google.com/issues/227803526.

**Comment by hao-nan-li** [comment]
Additional notes for the reviewer: No manual test added because in this case, there are 3 IAM tests generated from the example template.

File: mmv1/products/dns/api.yaml
Comment (roaks3) [inline mmv1/products/dns/api.yaml:44]: I'm not real familiar with the IAM-specific config here, but could you confirm that `managed_zone` is correct? I was expecting to see a field called `managed_zone` on this object, but didn't see one. The tests do appear to be passing, so I'm assuming everything works.

**Review by roaks3 — COMMENTED** [review]
LGTM, but I did have one question before giving approval

File: mmv1/products/dns/api.yaml
Comment (hao-nan-li) [reply mmv1/products/dns/api.yaml:44]: Thanks for reviewing, referring to https://googlecloudplatform.github.io/magic-modules/docs/how-to/add-mmv1-iam/.

The parent resource is called ManagedZone, which -> managed_zone

Comment (roaks3) [reply mmv1/products/dns/api.yaml:44]: Ah ok, that makes sense, thanks for the link!

**Review by roaks3 — APPROVED** [review]
Adding another team member for follow-up review

File: mmv1/templates/terraform/examples/dns_managed_zone_quickstart.tf.erb
Comment (melinath) [inline mmv1/templates/terraform/examples/dns_managed_zone_quickstart.tf.erb:41]: It looks like this is within the ci test project - in general we try to avoid that for non-ephemeral resources, but I'm not exactly sure where this _should_ go. Would it be possible to fix the configuration for `googlecloudexample.com` instead of setting up a new one? Alternately, would it be possible to generate new ones randomly here instead of hardcoding?

go/terraform-team-owned

**Review by melinath — CHANGES_REQUESTED** [review]
This looks good but it would be great to look into the test fix a little closer. Please also make sure the release note matches the release notes guide.

File: mmv1/templates/terraform/examples/dns_managed_zone_quickstart.tf.erb
Comment (hao-nan-li) [reply mmv1/templates/terraform/examples/dns_managed_zone_quickstart.tf.erb:41]: It looks like the priority of the failed tests pumped to P1. I need to read through some dns documentation and seek for a solution.

**Comment by hao-nan-li** [comment]
Got a problem coming up with git, anyway, I will split this PR into 2, as one of them adds the IAM resource, and another one to fix https://buganizer.corp.google.com/issues/260590932

**Pull request closed without merging**
